just tested the new pkg provided in the new pr. but got errors and exited immediately.

I have Nvidia 3090 GPU with the Nvidia driver, and the `glxgears` program runs OK.

both version = "0.2024.02.20.08.01.stable_02" and "_01" will cause the same issue.

outputs:
```
~ $ warp-terminal
/nix/store/xzhwd10mza8f0r51sadp0rfdp92rf08s-warp-terminal-0.2024.02.20.08.01.stable_02/opt/warpdotdev/warp-terminal/warp: /nix/store/84wcm74cp3k2i504ggwkgf4ikdyk0y3m-curl-8.6.0/lib/libcurl.so.4: no version information available (required by /nix/store/xzhwd10mza8f0r51sadp0rfdp92rf08s-warp-terminal-0.2024.02.20.08.01.stable_02/opt/warpdotdev/warp-terminal/warp)
stdout is a tty true, in CI false.  using logfile: false
14:46:12 [INFO] Spawning terminal server process...
/nix/store/xzhwd10mza8f0r51sadp0rfdp92rf08s-warp-terminal-0.2024.02.20.08.01.stable_02/opt/warpdotdev/warp-terminal/warp: /nix/store/84wcm74cp3k2i504ggwkgf4ikdyk0y3m-curl-8.6.0/lib/libcurl.so.4: no version information available (required by /nix/store/xzhwd10mza8f0r51sadp0rfdp92rf08s-warp-terminal-0.2024.02.20.08.01.stable_02/opt/warpdotdev/warp-terminal/warp)
14:46:12 [INFO] Running terminal server...
14:46:12 [WARN] Failed to load font Some("/nix/store/0i1hfpb7nglhq9ndp378gqwmda74a687-font-misc-misc-1.1.3/lib/X11/fonts/misc/6x13.pcf.gz"): InvalidFontFormat("PCF")
14:46:12 [WARN] Failed to load a font face 0 from source cause font doesn't have a family name.
14:46:12 [WARN] Unable to fetch system color scheme: FDO(
    UnknownMethod(
        "No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop",
    ),
)
14:46:12 [INFO] Failed to read User from secure storage NotFound
14:46:12 [INFO] Initializing crash reporting Some("stable_release") with tag "v0.2024.02.20.08.01.stable_02"...
14:46:12 [INFO] Starting warp with channel state ChannelState { channel: Stable, app_id: AppId { qualifier: "dev", organization: "warp", application_name: "Warp" }, additional_features: {}, firebase_api_key: "AIzaSyBdy3O3S9hrdayLJxJ7mriBR4qgUaUygAs", server_root_url: "https://app.warp.dev", ws_server_url: "wss://rtc.app.warp.dev/graphql", session_sharing_server_url: Some("wss://session-sharing-server-o3mgmiurkq-uk.a.run.app"), segment_write_key: "sTT9ZajzIaQ0phzLFOnqZ6VOqplJTlaJ", segment_root_url: "https://api.segment.io", releases_base_url: "https://releases.warp.dev", sentry_url: "https://0195a81da0714f55a93ee4624825f9ec@o540343.ingest.sentry.io/5658526", logfile_name: "warp.log", show_autoupdate_menu_items: true, skip_login: false } and version Some("v0.2024.02.20.08.01.stable_02")
14:46:12 [INFO] Start to flush telemetry events to Segment
14:46:12 [INFO] Performance metrics collector started
14:46:12 [INFO] Initializing app services
14:46:12 [WARN] No config found!
14:46:12 [WARN] No config found!
14:46:12 [INFO] Connecting to SQLite database
14:46:12 [INFO] Checking for update on channel stable_release. Update id is 10RmyrA
14:46:12 [INFO] fetching team tester status
14:46:12 [INFO] Fetching channel versions (without changelogs) from Warp server
14:46:12 [INFO] dispatching global action for root_view:open_from_restored
14:46:12 [INFO] dispatching global action for root_view:open_new
14:46:12 [WARN] Failed to load font Some("/nix/store/0i1hfpb7nglhq9ndp378gqwmda74a687-font-misc-misc-1.1.3/lib/X11/fonts/misc/6x13.pcf.gz"): InvalidFontFormat("PCF")
14:46:12 [WARN] Failed to load a font face 0 from source cause font doesn't have a family name.
14:46:12 [WARN] error getting team tester status: Failed to get access token for GraphQL request. Falling back to in-memory state.
14:46:12 [INFO] Guessed window scale factor: 1.5833333333333333
14:46:12 [WARN] No config found!
14:46:12 [WARN] No config found!
14:46:12 [INFO] Enabled wgpu backends: Backends(VULKAN | GL | METAL | DX12 | DX11 | BROWSER_WEBGPU)
14:46:12 [INFO] Available wgpu adapters:
14:46:12 [INFO] Other: NVIDIA GeForce RTX 3090/PCIe/SSE2
14:46:12 [INFO]         Driver: Unknown
14:46:12 [ERROR] Failed to open window: No wgpu adapter was found
14:46:12 [WARN] Redraw requested for a window for which we have no state
14:46:12 [WARN] Redraw requested for a window for which we have no state
14:46:12 [INFO] application will terminate
14:46:12 [INFO] Shutting down SQLite writer thread
14:46:12 [INFO] Shut down SQLite writer in 308.047µs
14:46:12 [INFO] Writing queued events to disk because telemetry is enabled.
14:46:12 [INFO] Successfully wrote telemetry events to disk
14:46:12 [INFO] Tearing down app services...
14:46:12 [INFO] Received empty message; assuming the connection has been closed.
```

```
~ $ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.1, NixOS, 24.05 (Uakari), 24.05.20240224.73de017`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
 - ```

--- a/pkgs/by-name/wa/warp-terminal/package.nix
+++ b/pkgs/by-name/wa/warp-terminal/package.nix
@@ -1,15 +1,74 @@
 { lib
 , stdenvNoCC
+, stdenv
 , fetchurl
+, autoPatchelfHook
 , undmg
+, zstd
+, curl
+, fontconfig
+, libglvnd
+, libxkbcommon
+, vulkan-loader
+, xdg-utils
+, xorg
+, zlib
 }:
-stdenvNoCC.mkDerivation (finalAttrs: {
-  pname = "warp-terminal";
-  version = "0.2023.12.05.08.02.stable_00";
 
+let
+pname = "warp-terminal";
+version = "0.2024.02.20.08.01.stable_01";
+
+linux = stdenv.mkDerivation (finalAttrs:  {
+  inherit pname version meta;
+  src = fetchurl {
+    url = "https://releases.warp.dev/stable/v${finalAttrs.version}/warp-terminal-v${finalAttrs.version}-1-x86_64.pkg.tar.zst";
+    hash = "sha256-L8alnqSE4crrDozRfPaAAMkLc+5+8d9XBKd5ddsxmD0=";
+  };
+
+  sourceRoot = ".";
+
+  postPatch = ''
+    substituteInPlace usr/bin/warp-terminal \
+      --replace-fail /opt/ $out/opt/
+  '';
+
+  nativeBuildInputs = [ autoPatchelfHook zstd ];
+
+  buildInputs = [
+    curl
+    fontconfig
+    stdenv.cc.cc.lib # libstdc++.so libgcc_s.so
+    zlib
+  ];
+
+  runtimeDependencies = [
+    libglvnd # for libegl
+    libxkbcommon
+    stdenv.cc.libc
+    vulkan-loader
+    xdg-utils
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcursor
+    xorg.libXi
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
+    cp -r opt usr/* $out
+
+    runHook postInstall
+  '';
+});
+
+darwin = stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit pname version meta;
   src = fetchurl {
     url = "https://releases.warp.dev/stable/v${finalAttrs.version}/Warp.dmg";
-    hash = "sha256-9olAmczIPRXV15NYCOYmwuEmJ7lMeaQRTTfukaYXMR0=";
+    hash = "sha256-tFtoD8URMFfJ3HRkyKStuDStFkoRIV97y9kV4pbDPro=";
   };
 
   sourceRoot = ".";
@@ -24,13 +83,18 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
     runHook postInstall
   '';
+});
 
-  meta = with lib; {
-    description = "Rust-based terminal";
-    homepage = "https://www.warp.dev";
-    license = licenses.unfree;
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    maintainers = with maintainers; [ emilytrau Enzime ];
-    platforms = platforms.darwin;
-  };
-})
+meta = with lib; {
+  description = "Rust-based terminal";
+  homepage = "https://www.warp.dev";
+  license = licenses.unfree;
+  sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+  maintainers = with maintainers; [ emilytrau Enzime ];
+  platforms = platforms.darwin ++ [ "x86_64-linux" ];
+};
+
+in
+if stdenvNoCC.isDarwin
+then darwin
+else linux

